### PR TITLE
i-bem-dom: update _elemCache on findElem

### LINKED ISSUE #583: i-bem, i-bem-dom: `findElem` should update elem's cache
Can `findElem` method rewrite `elem`'s internal cache if called with the same parameters?

### PR DESCRIPTION
**Pull request by qfox**
Fixes #583

### CODE REVIEW

File: common.blocks/i-bem-dom/i-bem-dom.js [hunk 1]
```diff
@@ -768,7 +773,7 @@ var Block = inherit(BEM.Block,/** @lends Block.prototype */{
 
         var key = elem.elem + buildModPostfix(elem.modName, elem.modVal);
```
Comment (qfox) [inline common.blocks/i-bem-dom/i-bem-dom.js]: Может это тоже унести в findElem и там оставить проверку, если нет ctx?

Comment (veged) [inline common.blocks/i-bem-dom/i-bem-dom.js]: мы же не сможем тогда построить ключ для кеша, который нужен тут

Comment (qfox) [inline common.blocks/i-bem-dom/i-bem-dom.js]: Да, уже понял ;-(

**Comment by narqo** [comment]
@zxqfox what's up!? How is going?

**Comment by qfox** [comment]
@narqo Waiting for #842 ;-)

**Comment by narqo** [comment]
We've merged #842 What's next?

File: common.blocks/i-bem-dom/i-bem-dom.js [hunk 2]
```diff
@@ -740,6 +740,11 @@ var Block = inherit(BEM.Block,/** @lends Block.prototype */{
         var res = findDomElem(ctx, this.buildSelector(elem.elem, elem.modName, elem.modVal));
         res.__bemElemName = elem.elem;
 
+        if(ctx === this.domElem) {
+            var key = elem.elem + buildModPostfix(elem.modName, elem.modVal);
```
Comment (narqo) [inline common.blocks/i-bem-dom/i-bem-dom.js]: As `key` is used only once, it can be omited

**Comment by qfox** [comment]
Need to rebase ;-\

**Comment by qfox** [comment]
@narqo Omfg. It's impossible to rebase! There are no `findElem` method now!

**Comment by narqo** [comment]
> [..] There are no findElem method now

Yep! The API has got some improvements in `v3`. See #581 for reference

**Comment by qfox** [comment]
@narqo Since it can't be ported as is because business logic was changed. Am I right to add it to findChild\* methods (and not the rest)?

Time to take a look ;-)

File: common.blocks/i-bem-dom/i-bem-dom.spec.js [hunk 3]
```diff
@@ -713,6 +742,45 @@ describe('i-bem-dom', function() {
...
+            it('should update _elemsCache on findChildElem call', function() {
+                b1Block.elems('e1');
+                rootNode.html(BEMHTML.apply({
+                    block : 'b1',
+                    elem : 'e1',
+                    js : { id : '1``' }
+                }));
+                b1Block.findChildElems('e1');
+
+                getEntityIds(b1Block.elems('e1')).should.be.eql(['1``']);
```
Comment (dfilatov) [inline common.blocks/i-bem-dom/i-bem-dom.spec.js]: Why do you use "``" symbols here?

Comment (qfox) [inline common.blocks/i-bem-dom/i-bem-dom.spec.js]: It's just a POC. to differentiate e1 without mods with ones with them.

Comment (qfox) [inline common.blocks/i-bem-dom/i-bem-dom.spec.js]: @dfilatov Should I change them all to numbers?

**Comment by qfox** [comment]
@narqo @dfilatov Anything to do here?

File: common.blocks/i-bem-dom/i-bem-dom.spec.js (hunk 3 above)
Comment (dfilatov) [inline common.blocks/i-bem-dom/i-bem-dom.spec.js]: I think you should )

**Comment by qfox** [comment]
@dfilatov Done

**Pull request merged**